genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/JobFolderArchiveServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services.impl;
+
+import com.netflix.genie.agent.execution.ExecutionContext;
+import com.netflix.genie.agent.execution.exceptions.ArchiveException;
+import com.netflix.genie.agent.execution.services.JobFolderArchiveService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+/**
+ * Default implementation of JobFolderArchiveService.
+ * Does no archive. Provide an implementation for archiving if needed
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@Component
+@Lazy
+@Slf4j
+class JobFolderArchiveServiceImpl implements JobFolderArchiveService {
+
+    /**
+     * Default implementation does not archive.
+     *
+     * @param executionContext  Execution context
+     * @throws ArchiveException In case of a problem
+     */
+    @Override
+    public void archiveJobFolder(final ExecutionContext executionContext) throws ArchiveException {
+        log.info("Job folder not archived. Please provide an implementation if you wish to archive.");
+    }
```

Comment:
The way I am thinking about this is that this default implementation will not do any archival. There will be a s3 based implementation provided for Netflix use case which will provide a S3 URL to the uploadService. UploadService will then compress and upload it to the S3 location